blog/2024-12-29-rust-gpu-0.10.md

```diff
@@ -0,0 +1,250 @@
+
+## What's new in 0.10
+
+### Update Rust requirement to a modern version
+
+Rust GPU includes a compiler backend that compiles regular Rust code into
+[SPIR-V](https://www.khronos.org/spir/), a low-level format that [most GPUs
+understand](https://vulkan.gpuinfo.org/). Because of this deep integration with compiler
+internals, Rust GPU must use a very specific version of the Rust compiler. Rust GPU now
+supports `nightly-2024-11-22` (up from the ancient `nightly-2023-05-27`). This roughly
+corresponds to [Rust 1.83](https://blog.rust-lang.org/2024/11/28/Rust-1.83.0.html).
+
+:::tip
+
+Most real-world projects like [`krnl`](https://github.com/charles-r-earp/krnl) and
+[`renderling`](https://github.com/schell/renderling) compile their GPU code with Rust
+GPU's specific version of nightly while using stable Rust for the rest of their code. We
+are in the process of making this workflow much easier.
+
+:::
+
+Updating to a newer nightly version required heroic effort from
+[@eddyb](https://github.com/eddyb), as significant breaking changes to Rust's [internal
+allocation model](https://github.com/rust-lang/rust/pull/122053) and
+[`#[repr(simd)]`](https://github.com/rust-lang/rust/pull/129403) were introduced in
+`rustc`. [@eddyb](https://github.com/eddyb) was able to [vendor in parts of `rustc` and
+patch out others](https://github.com/Rust-GPU/rust-gpu/pull/170), unblocking the update
+and buying us time to [figure out the best way
+forward](https://github.com/Rust-GPU/rust-gpu/issues/182).
+
+Rust GPU is an "out-of-tree" compiler backend. This enables fast iteration while we
+explore the problem space, but also makes us more susceptible to this sort of breakage.
+One of our long-term goals is to be a supported `rustc` backend.
+
+### **Mesh shader support**
+
+Rust GPU maintainer [@Firestar99](https://github.com/firestar99) and contributor
+[@BeastLe9enD](https://github.com/BeastLe9enD) added support for [mesh
+shaders](https://github.com/Rust-GPU/rust-gpu/pull/44).
+
+[Mesh shaders](https://www.khronos.org/blog/mesh-shading-for-vulkan) are a modern GPU
+feature that replace traditional vertex and geometry shaders. They give GPU developers
+more control over the entire pipeline by letting them directly process batches of
+vertices and primitives in compute-like shaders. This allows for more efficient culling,
+level-of-detail calculations, and custom pipeline logic—all on the GPU.
+
+An example of a Rust GPU mesh shader that outputs points:
+
+```
+use spirv_std::arch::set_mesh_outputs_ext;
+use spirv_std::glam::{UVec2, Vec4};
+use spirv_std::spirv;
+
+#[spirv(mesh_ext(
+    threads(1),
+    output_vertices = 1,
+    output_primitives_ext = 1,
+    output_points
+))]
+pub fn main(
+    #[spirv(position)] positions: &mut [Vec4; 1],
+    #[spirv(primitive_point_indices_ext)] indices: &mut [u32; 1],
+) {
+    unsafe {
+        set_mesh_outputs_ext(1, 1);
+    }
+
+    positions[0] = Vec4::new(-0.5, 0.5, 0.0, 1.0);
+
+    indices[0] = 0;
+}
+```
```

Comment:
I'd use the triangle example instead, barely anyone uses points or lines and it's just 2 lines longer. 
https://github.com/Rust-GPU/rust-gpu/blob/710e394df7078125468b5063b3ceecabccb5bd8f/tests/ui/arch/mesh_shader_output_triangles.rs

Should we remove the imports?

Reply:
Ok, changed! I like keeping the imports because it not only makes it complete but shows we have a module system. There aren't too many so I don't think they are too distracting but happy to change if you feel strongly.